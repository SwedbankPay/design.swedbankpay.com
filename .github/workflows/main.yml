name: Main
on:
  push:
    branches:
      - "feature/**"
      - "develop"
      - "release/**"

jobs:
  build:
    name: Build and deploy
    runs-on: ubuntu-latest
    strategy:
      matrix:
        brand: ["payex", "swedbankpay"]
    steps:
      - uses: actions/checkout@v4

      # Set brand specific variables
      - name: Environment variables
        id: variables
        run: ./.github/scripts/variables.sh --brand ${{ matrix.brand }} --ref ${{ github.ref }}

      - name: Set Node.js 18.x
        uses: actions/setup-node@v4
        with:
          node-version: 18.x

      - name: Cache dependencies
        uses: actions/cache@v4
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: ${{ runner.os }}-node-

      - name: Install
        run: npm ci

      - name: Build
        run: npm run ${{ steps.variables.outputs.BUILD_SCRIPT }} -- --env basename=${{ steps.variables.outputs.BRANCH }} --env semver=${{ steps.variables.outputs.BRANCH }} --env info_version=${{ steps.variables.outputs.VERSION }} --env github_actions=true

      - name: Create latest version reference file
        run: |
          echo "{ \"latestVersion\" : \"${{ steps.variables.outputs.BRANCH }}\" }" > ./dist/latestVersion.json

      - name: Azure Login
        uses: azure/login@v2.3.0
        with:
          creds: ${{ secrets.AZURE_CREDENTIALS_STAGE }}

      - name: Azure Deploy
        uses: azure/CLI@v2
        with:
          azcliversion: 2.43.0
          inlineScript: |
            az storage blob upload-batch -s dist -d \$web --account-name ${{ steps.variables.outputs.AZURE_ACCOUNT_STAGE }} --overwrite
        env:
          AZURE_CORE_USE_MSAL_HTTP_CACHE: false
