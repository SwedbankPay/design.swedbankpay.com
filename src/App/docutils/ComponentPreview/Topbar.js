import React from "react";

export const Topbar = ({ tabsId, tabs, hideOptions, onTabClick,
    sidebarDisplayed, setSidebarDisplayed, setActiveTab, activeTab }) => (
    <div id={tabsId} className="tabs tabs-scroll">
        { tabs && <Tabs
            tabsId={tabsId}
            tabs={tabs}
            onTabClick={onTabClick}
            activeTab={activeTab}
            setActiveTab={setActiveTab}
        />}
        <CloseOptionsButton
            hideOptions={hideOptions}
            sidebarDisplayed={sidebarDisplayed}
            setSidebarDisplayed={setSidebarDisplayed}

        />
    </div>
);

/* TODO: switch aanchor links to buttons */
const Tabs = ({ tabsId, tabs, activeTab, setActiveTab, onTabClick }) => (
    <ul id={`${tabsId}-ul`}>
        {tabs.map((element, i) => <li key={i} className={`${activeTab?.tab === element.tab ? "active" : null}`}>
            <a href="#" onClick={e => setActiveTab(tabs[i])}>{element.tab}</a>
        </li>
        )}
    </ul>
);

const CloseOptionsButton = ({ hideOptions, sidebarDisplayed, setSidebarDisplayed }) => (
    <div className={`options-open ${hideOptions ? " d-none" : ""} ${sidebarDisplayed ? "hidden" : ""}`}>
        <button
            className="open-options-menu d-flex"
            aria-label="Open options menu"
            onClick={() => setSidebarDisplayed(true)}
        >
            <i className="material-icons" >menu_open</i>
        </button>
    </div>
);

