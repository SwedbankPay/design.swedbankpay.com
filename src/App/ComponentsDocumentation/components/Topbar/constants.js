import React, { useEffect } from "react";
import TopbarComponent from "@components/Topbar";
import { topbar } from "@src/scripts/main";

import CodeTags from "@components/CodeTags";

export const menu = {
	items: [
		{
			name: "Home",
			pinned: true,
		},
		{
			name: "Merchant details",
			selected: true,
			pinned: true,
		},
		{
			name: "Transactions",
		},
		{
			name: "User profile",
		},
		{
			name: "Statistics",
		},
		{
			name: "Transactions",
		},
		{
			name: "User profile",
		},
		{
			name: "Statistics",
		},
		{
			name: "Transactions",
		},
		{
			name: "User profile",
		},
		{
			name: "Statistics",
		},
		{
			name: "Users",
			pinned: true,
		},
		{
			name: "Access tokens",
		},
		{
			name: "Contact us",
		},
		{
			name: "Switch to acquiring",
			icon: "sync_alt",
			pinned: true,
			firstPushRight: true,
		},
		{
			name: "Change company",
			icon: "arrow_forward",
		},
	],
};

export const menuLegacy = {
	btn: {
		name: "Menu",
		icon: "menu",
	},
	items: [
		{
			name: "Home",
			icon: "home",
		},
		{
			name: "Purchases",
			icon: "shopping_cart",
		},
		{
			name: "Settings",
			icon: "settings",
		},
	],
};

const Topbar = ({ sticky, wide, logout, legacy }) => {
	useEffect(() => {
		topbar.init();
	}, [legacy]);

	return (
		<div className="container-fluid">
			<TopbarComponent
				topbarContent={!legacy ? menu : menuLegacy}
				wide={wide}
				sticky={sticky}
				logout={logout}
				id="demo-topbar"
				legacy={legacy}
			/>
			<main className="topbar-content"></main>
		</div>
	);
};

export const topbarShowcase = {
	id: "overviewTopbar",
	tabsId: "overviewTopbarTabs",
	hideOptions: true,
	elements: [
		{
			tab: "Desktop",
			component: <Topbar wide="xl" logout legacy={true} />,
			title: "Desktop",
			description: (
				<p>
					The minimum requirement in a Topbar is to include the Swedbank Pay
					Logotype, common additional functionality is a list with navigation
					links. On desktop use{" "}
					<CodeTags type="secondary" code=".topbar-xl-wide" /> to show the links
					listed horizontally in the topbar.{" "}
				</p>
			),
		},
		{
			tab: "Mobile/tablet",
			component: <Topbar logout wide={false} legacy={true} />,
			title: "Mobile/tablet",
			description: (
				<>
					<p>
						The minimum requirement in a Topbar is to include the Swedbank Pay
						Logotype, common additional functionality is a list with navigation
						links. On smaller screens use a menu button to toggle a vertical
						navigation drawer with links when the menu button is clicked.
					</p>
					<p>
						Be aware; The <CodeTags code="max-height" type="primary" />{" "}
						attribute for element <CodeTags type="secondary" code=".nav-ul" />{" "}
						when it is active is set to 2500px. This is to ensure animation for
						the transition to happen. You can easily alter this by creating
						custom css.
					</p>
				</>
			),
		},
		{
			tab: "ðŸ§ª experimental Desktop",
			component: <Topbar wide="xl" logout legacy={false} />,
			title: "Desktop",
			description: (
				<p>
					The experimental desktop topbar involves breaking changes. Make sure
					to check them out before switching.
				</p>
			),
		},
		{
			tab: "ðŸ§ª experimental Mobile/tablet",
			component: <Topbar logout wide={false} legacy={false} />,
			title: "Mobile/tablet",
			description: (
				<>
					The experimental desktop topbar involves breaking changes. Make sure
					to check them out before switching.
				</>
			),
		},
	],
};
