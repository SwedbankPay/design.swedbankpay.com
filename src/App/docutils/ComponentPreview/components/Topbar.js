import React, { useEffect } from "react";
import { tabs as tabsScript } from "@src/scripts/main";

export const Topbar = ({
	tabsId,
	tabs,
	hideOptions,
	sidebarDisplayed,
	setSidebarDisplayed,
	updateActiveTab,
	activeTab,
}) => (
	<div id={tabsId} className="tabs tabs-scroll">
		{tabs && (
			<Tabs
				tabsId={tabsId}
				tabs={tabs}
				activeTab={activeTab}
				updateActiveTab={updateActiveTab}
			/>
		)}
		<CloseOptionsButton
			hideOptions={hideOptions}
			sidebarDisplayed={sidebarDisplayed}
			setSidebarDisplayed={setSidebarDisplayed}
		/>
	</div>
);

const Tabs = ({ tabsId, tabs, activeTab, updateActiveTab }) => {
	useEffect(() => {
		tabsScript.init();
	}, []);

	return (
		<ul id={`${tabsId}-ul`}>
			{tabs.map((element, i) => (
				<li
					key={i}
					className={`${activeTab?.tab === element.tab ? "active" : null}`}
				>
					<a
						href="#"
						onClick={(e) => {
							e.preventDefault();
							updateActiveTab(tabs[i]);
						}}
					>
						{element.tab}
					</a>
				</li>
			))}
		</ul>
	);
};

const CloseOptionsButton = ({
	hideOptions,
	sidebarDisplayed,
	setSidebarDisplayed,
}) => (
	<div
		className={`options-open ${hideOptions ? " d-none" : ""} ${
			sidebarDisplayed ? "hidden" : ""
		}`}
	>
		<button
			className="open-options-menu d-flex"
			aria-label="Open options menu"
			onClick={() => setSidebarDisplayed(true)}
		>
			<i className="material-icons">menu_open</i>
		</button>
	</div>
);
