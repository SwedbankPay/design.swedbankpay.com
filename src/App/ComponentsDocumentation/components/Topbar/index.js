import React, { Component } from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, DocContainer, JavascriptDocs } from "@docutils";
import TopbarComponent from "@components/Topbar";
import CodeTags from "@components/CodeTags";
import SidebarComponent from "@components/Sidebar";
import { SidebarNavList } from "../Sidebar/constants";

const { topbar, sidebar } = window.dg;

const menu = {
    btn: {
        name: "Menu",
        icon: "menu"
    },
    items: [
        {
            name: "Home",
            icon: "home"
        },
        {
            name: "Purchase history",
            icon: "shopping_cart"
        },
        {
            name: "Settings",
            icon: "settings"
        }
    ]
};

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            To use a topbar the bare minimum markup required is:
        </p>
        <ul>
            <li><CodeTags type="secondary" code=".topbar" /> wrapper.</li>
            <li>Anchor with class <CodeTags type="secondary" code=".topbar-logo" /> to display the logo and make it clickable.</li>
        </ul>
        <p>
            With this you get a topbar with the Swedbank Pay logo.
            In the topbar you can add additional functionality like a menu containing links. To enable this add a <CodeTags type="primary" code={"<button>"} /> and
            a <CodeTags type="primary" code={"<nav>"} /> containing anchors. Remember to include <CodeTags type="secondary" code={"toggle-nav=\"{your_nav_id}\""} /> as
            a <CodeTags type="primary" code={"<button>"} /> attribute
            to get the menu to appear when the menu-button is clicked.
        </p>

        <p>
            Note that scrolling is locked when the menu is displayed.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout id="demo-topbar" />
        </ComponentPreview>
    </>
);

const TopbarWide = () => (
    <>
        <h2 id="topbar-wide">Wide topbar</h2>
        <p>
            This is a wide topbar.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent id="demo-topbar-1" topbarContent={menu} wide="xl" logout />
        </ComponentPreview>
    </>
);

const TopbarSticky = () => (
    <>
        <h2 id="topbar-sticky">Sticky topbar</h2>
        <p>
            A sticky topbar is also available, with class <CodeTags type="secondary" code=".topbar-sticky" />.
            Use <CodeTags type="secondary" code=".topbar-min-{breakpoint}-sticky" /> to specify the minimum width the topbar should be sticky,
            or <CodeTags type="secondary" code=".topbar-max-{breakpoint}-sticky" /> to specify the maximum width the topbar should be sticky
            (see <Link to="/utilities/breakpoints">breakpoints</Link> for more information on breakpoints). Note that the sticky topbar does
            not work in Internet Explorer 11.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="container-fluid">
                <TopbarComponent id="demo-topbar-sticky" topbarContent={menu} wide="xl" logout sticky />
                <div className="row">
                    <div className="col-auto">
                        <SidebarComponent id="topbar-sticky-sidebar" sidebarNavList={SidebarNavList} sticky />
                    </div>
                    <main className="col main-content border-left">Main content things...</main>
                </div>
            </div>
        </ComponentPreview>
    </>
);

const TopbarWithSidebar = () => (
    <>
        <h2 id="topbar-with-sidebar">Topbar with sidebar content</h2>
        <p>
            It is recommended to have the sidebar content in the topbar menu on smaller devices, such as phones and tablets (see more on sidebar <Link to="/components/sidebar">here</Link>).
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout id="demo-topbar-sidebar" sidebar />
        </ComponentPreview>
    </>
);

const TopbarPNG = () => (
    <>
        <h2 id="topbar-png">Topbar with PNG logo</h2>
        <p>
            A PNG version of the logo can also be used. This is done by appending the class <CodeTags type="secondary" code=".topbar-logo-png" /> to the
            anchor with class <CodeTags type="secondary" code=".topbar-logo" />. Only consider using the PNG version when there are issues with using the SVG version.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout id="demo-topbar-png" png/>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent id="demo-topbar-1-png" topbarContent={menu} wide="xl" logout png/>
        </ComponentPreview>
    </>
);

const JavascriptMethods = () => (
    <>
        <h2 id="javascript-methods">JavaScript Methods</h2>
        <JavascriptDocs componentName="topbar" open close />
    </>
);

class Topbar extends Component {
    componentDidMount () {
        topbar.init();
        sidebar.init("topbar-sidebar");
        sidebar.init("topbar-sticky-sidebar");
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">The topbar is used to give users an easily available navigational bar at the top of your web application.</p>
                <Overview />
                <TopbarWide />
                <TopbarSticky />
                <TopbarWithSidebar />
                <TopbarPNG />
                <JavascriptMethods />
            </DocContainer>
        );
    }
}

export default Topbar;

/* For testing */
export { Overview, TopbarWide, TopbarSticky, TopbarWithSidebar, TopbarPNG, JavascriptMethods };
