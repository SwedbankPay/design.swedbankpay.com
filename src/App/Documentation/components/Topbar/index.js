import React, { Component } from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, DocContainer, JavascriptDocs } from "@docutils";
import TopbarComponent from "@components/Topbar";
import CodeTags from "@components/CodeTags";
import SidebarComponent from "@components/Sidebar";
import { SidebarNavList } from "../Sidebar/constants";

const { topbar } = window.dg;

const menu = {
    btn: {
        name: "Menu",
        icon: "menu"
    },
    items: [
        {
            name: "Home",
            icon: "home"
        },
        {
            name: "Purchase history",
            icon: "shopping_cart"
        },
        {
            name: "Settings",
            icon: "settings"
        }
    ]
};

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            To use a topbar the bare minimum markup required is:
        </p>
        <ul>
            <li><CodeTags type="secondary" code=".topbar" /> wrapper.</li>
            <li>Anchor with class <CodeTags type="secondary" code=".topbar-logo" /> to display the logo and make it clickable.</li>
        </ul>
        <p>
            With this you get a topbar with the Swedbank Pay logo.
            In the topbar you can add additional functionality like a menu containing links. To enable this add a <CodeTags type="primary" code={"<button>"} /> and
            a <CodeTags type="primary" code={"<nav>"} /> containing anchors. Remember to include <CodeTags type="secondary" code={"toggle-nav=\"{your_nav_id}\""} /> as
            a <CodeTags type="primary" code={"<button>"} /> attribute
            to get the menu to appear when the menu-button is clicked.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout id="demo-topbar" />
        </ComponentPreview>
    </>
);

const TopbarWide = () => (
    <>
        <h2 id="topbar-wide">Wide topbar</h2>
        <p>
            This is a wide topbar.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent id="demo-topbar-1" topbarContent={menu} wide="xl" logout />
        </ComponentPreview>
    </>
);

const TopbarSticky = () => (
    <>
        <h2 id="topbar-sticky">Sticky topbar</h2>
        <p>
            A sticky topbar is also available, with class <CodeTags type="secondary" code=".topbar-sticky" />.
            Use <CodeTags type="secondary" code=".topbar-min-{breakpoint}-sticky" /> to specify the minimum width the topbar should be sticky,
            or <CodeTags type="secondary" code=".topbar-max-{breakpoint}-sticky" /> to specify the maximum width the topbar should be sticky
            (see <Link to="/docs/core/breakpoints">breakpoints</Link> for more information on breakpoints).
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="container-fluid">
                <TopbarComponent id="demo-topbar-sticky" topbarContent={menu} wide="xl" logout sticky />
                <div className="row">
                    <div className="col-auto">
                        <SidebarComponent id="dg-sidebar" sidebarNavList={SidebarNavList} sticky />
                    </div>
                    <main className="col main-content border-left">Main content things...</main>
                </div>
            </div>
        </ComponentPreview>
    </>
);

const TopbarPNG = () => (
    <>
        <h2 id="topbar-png">Topbar with PNG logo</h2>
        <p>
            A PNG version of the logo can also be used. This is done by appending the class <CodeTags type="secondary" code=".topbar-logo-png" /> to the
            anchor with class <CodeTags type="secondary" code=".topbar-logo" />. Only consider using the PNG version when there are issues with using the SVG version.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout id="demo-topbar-png" png/>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent id="demo-topbar-1-png" topbarContent={menu} wide="xl" logout png/>
        </ComponentPreview>
    </>
);

const JavascriptMethods = () => (
    <>
        <h2 id="javascript-methods">JavaScript Methods</h2>
        <JavascriptDocs componentName="topbar" open close />
    </>
);

class Topbar extends Component {
    componentDidMount () {
        topbar.init("demo-topbar");
        topbar.init("demo-topbar-1");
        topbar.init("demo-topbar-png");
        topbar.init("demo-topbar-1-png");
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">The topbar is used to give users an easily available navigational bar at the top of your web application.</p>
                <Overview />
                <TopbarWide />
                <TopbarSticky />
                <TopbarPNG />
                <JavascriptMethods />
            </DocContainer>
        );
    }
}

export default Topbar;

/* For testing */
export { Overview, TopbarWide, TopbarSticky, TopbarPNG, JavascriptMethods };
