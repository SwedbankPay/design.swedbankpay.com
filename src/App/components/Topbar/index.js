import React, { Fragment } from "react";
import PropTypes from "prop-types";
import swedbankpayLogo from "@src/img/swedbankpay/logo/swedbankpay-logo-v.svg";
import payexLogo from "@src/img/payex/logo/payex-logo.svg";

import SidebarComponent from "@components/Sidebar";
import { SidebarNavList } from "../../ComponentsDocumentation/components/Sidebar/constants";

const brand = process.env.brand;

const devLogo = brand === "swedbankpay" ? swedbankpayLogo : payexLogo;

const isDev = process.env.version === "LOCAL_DEV";

const TopbarBtn = () => (
    <>
        <button type="button" className="topbar-btn">{"\n\t\t"}
            <i className="material-icons topbar-btn-icon" aria-hidden="true">menu</i>
            {"\n\t\t"}
        </button>
        <button type="button" className="topbar-close">{"\n"}
            <i className="material-icons topbar-btn-icon" aria-hidden="true">close</i>{"\n"}
            {"\n\t\t"}
        </button>
    </>
);

const TopbarMenu = ({ menu, logout, sidebar }) => {
    const { items } = menu;

    return (
        <nav className="topbar-nav">
            <div className="topbar-link-container">{"\n"}
                {items.map((item, i) => (
                    <Fragment key={i}>
                        {"\n"}
                        <a href="#" className={item.name === "Home" ? "active" : null} onClick={e => e.preventDefault()}>
                            {"\n"}
                            {item.icon ? <i className="material-icons" aria-hidden="true">{item.icon}</i> : null}
                            {"\n"}
                            <span>{item.name}</span>
                            {"\n"}
                        </a>
                        {i === 0 && sidebar && <SidebarComponent id="topbar-sidebar" sidebarNavList={SidebarNavList} />}
                    </Fragment>
                ))}{"\n"}
                {logout ? <TopbarLogout /> : null}
            </div>
        </nav>
    );
};

const TopbarLogout = () => (
    <>
        <a className="topbar-link-right" href="#" onClick={e => e.preventDefault()}>{"\n"}
            <i className="material-icons" aria-hidden="true">exit_to_app</i>{"\n"}
            <span>Log out</span>{"\n"}
        </a>{"\n"}
    </>
);

const TopbarLogo = ({ png }) => (
    <>
        <a className={`topbar-logo${png ? " topbar-logo-png" : ""}`} href="/" onClick={e => e.preventDefault()}>{"\n"}
            {(brand === "swedbankpay" && png)
                ? <img src={`${process.env.basename}img/${brand}-logo${brand === "swedbankpay"
                    ? "-v"
                    : ""}.png`} alt={`${brand}-logo`} className="logotype-vertical logotype-md"/>
                : <img src={isDev
                    ? devLogo
                    : `${process.env.basename}img/${brand}-logo${brand === "swedbankpay"
                        ? "-v"
                        : ""}.svg`} alt={`${brand}-logo`} className="logotype-vertical logotype-md"/>
            }{"\n"}
        </a>{"\n"}
    </>
);

const Topbar = ({ topbarContent, wide, logout, id, png, sticky, sidebar }) => (
    <header className={`topbar${wide ? ` topbar-${wide}-wide` : ""}${sticky ? " topbar-sticky" : ""}`} id={id}>{"\n"}
        {topbarContent ?
        <>
            <TopbarBtn />{"\n"}
            <TopbarLogo png={png}/>
            <TopbarMenu menu={topbarContent} logout={!!logout} sidebar={sidebar} />
        </> :
        <>
            <TopbarLogo png={png} />
            {logout ? <TopbarLogout /> : null}
        </>}
    </header>
);

Topbar.propTypes = {
    topbarContent: PropTypes.object,
    fixed: PropTypes.bool,
    logout: PropTypes.bool,
    png: PropTypes.bool,
    sticky: PropTypes.bool,
    sidebar: PropTypes.bool
};

export default Topbar;
