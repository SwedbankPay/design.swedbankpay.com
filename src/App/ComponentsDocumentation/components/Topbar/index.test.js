import React from "react";
import { shallow } from "enzyme";

import Topbar, { Overview, DeveloperDocumentation } from "./index";

import { topbar } from "@src/scripts/main";

describe("Components: Topbar", () => {
    it("is defined", () => {
        expect(Topbar).toBeDefined();
    });

    it("renders", () => {
        /*
            The Topbar component calls dg.topbar.init("demo-topbar") on componentDidMount. Mocking the topbarscript
            to avoid a warning message. [AW]
        */
        topbar.init = jest.fn();

        const wrapper = shallow(<Topbar />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("DeveloperDocumentation", () => {
        it("is defined", () => {
            expect(DeveloperDocumentation).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<DeveloperDocumentation />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
