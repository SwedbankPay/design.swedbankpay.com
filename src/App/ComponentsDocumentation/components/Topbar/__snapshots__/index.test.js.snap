// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Topbar JavascriptMethods renders 1`] = `
<Fragment>
  <h2
    id="javascript-methods"
  >
    JavaScript Methods
  </h2>
  <JavascriptDocs
    close={true}
    componentName="topbar"
    open={true}
  />
</Fragment>
`;

exports[`Components: Topbar Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    To use a topbar the bare minimum markup required is:
  </p>
  <ul>
    <li>
      <CodeTags
        code=".topbar"
        type="secondary"
      />
       wrapper.
    </li>
    <li>
      Anchor with class 
      <CodeTags
        code=".topbar-logo"
        type="secondary"
      />
       to display the logo and make it clickable.
    </li>
  </ul>
  <p>
    With this you get a topbar with the Swedbank Pay logo. In the topbar you can add additional functionality like a menu containing links. To enable this add a 
    <CodeTags
      code="<button>"
      type="primary"
    />
     and a 
    <CodeTags
      code="<nav>"
      type="primary"
    />
     containing anchors. Remember to include 
    <CodeTags
      code="toggle-nav=\\"{your_nav_id}\\""
      type="secondary"
    />
     as a 
    <CodeTags
      code="<button>"
      type="primary"
    />
     attribute to get the menu to appear when the menu-button is clicked.
  </p>
  <p>
    Note that scrolling is locked when the menu is displayed.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar"
      logout={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "name": "Menu",
          },
          "items": Array [
            Object {
              "icon": "home",
              "name": "Home",
            },
            Object {
              "icon": "shopping_cart",
              "name": "Purchase history",
            },
            Object {
              "icon": "settings",
              "name": "Settings",
            },
          ],
        }
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar TopbarPNG renders 1`] = `
<Fragment>
  <h2
    id="topbar-png"
  >
    Topbar with PNG logo
  </h2>
  <p>
    A PNG version of the logo can also be used. This is done by appending the class 
    <CodeTags
      code=".topbar-logo-png"
      type="secondary"
    />
     to the anchor with class 
    <CodeTags
      code=".topbar-logo"
      type="secondary"
    />
    . Only consider using the PNG version when there are issues with using the SVG version.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar-png"
      logout={true}
      png={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "name": "Menu",
          },
          "items": Array [
            Object {
              "icon": "home",
              "name": "Home",
            },
            Object {
              "icon": "shopping_cart",
              "name": "Purchase history",
            },
            Object {
              "icon": "settings",
              "name": "Settings",
            },
          ],
        }
      }
    />
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar-1-png"
      logout={true}
      png={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "name": "Menu",
          },
          "items": Array [
            Object {
              "icon": "home",
              "name": "Home",
            },
            Object {
              "icon": "shopping_cart",
              "name": "Purchase history",
            },
            Object {
              "icon": "settings",
              "name": "Settings",
            },
          ],
        }
      }
      wide="xl"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar TopbarSticky renders 1`] = `
<Fragment>
  <h2
    id="topbar-sticky"
  >
    Sticky topbar
  </h2>
  <p>
    A sticky topbar is also available, with class 
    <CodeTags
      code=".topbar-sticky"
      type="secondary"
    />
    . Use 
    <CodeTags
      code=".topbar-min-{breakpoint}-sticky"
      type="secondary"
    />
     to specify the minimum width the topbar should be sticky, or 
    <CodeTags
      code=".topbar-max-{breakpoint}-sticky"
      type="secondary"
    />
     to specify the maximum width the topbar should be sticky (see 
    <Link
      to="/utilities/breakpoints"
    >
      breakpoints
    </Link>
     for more information on breakpoints). Note that the sticky topbar does not work in Internet Explorer 11.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="container-fluid"
    >
      <Topbar
        id="demo-topbar-sticky"
        logout={true}
        sticky={true}
        topbarContent={
          Object {
            "btn": Object {
              "icon": "menu",
              "name": "Menu",
            },
            "items": Array [
              Object {
                "icon": "home",
                "name": "Home",
              },
              Object {
                "icon": "shopping_cart",
                "name": "Purchase history",
              },
              Object {
                "icon": "settings",
                "name": "Settings",
              },
            ],
          }
        }
        wide="xl"
      />
      <div
        className="row"
      >
        <div
          className="col-auto"
        >
          <Sidebar
            id="topbar-sticky-sidebar"
            sidebarNavList={
              Array [
                Object {
                  "icon": <i
                    className="material-icons-outlined"
                  >
                    home
                  </i>,
                  "title": "Home Page",
                },
                Object {
                  "children": Array [
                    Object {
                      "icon": <i
                        className="material-icons-outlined rotated"
                      >
                        accessibility
                      </i>,
                      "title": "Accessibility",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "title": "Primary colors",
                        },
                        Object {
                          "title": "Secondary colors",
                        },
                        Object {
                          "title": "Tertiary colors",
                        },
                      ],
                      "icon": <i
                        className="material-icons-outlined rotated"
                      >
                        palette
                      </i>,
                      "title": "Colors",
                    },
                    Object {
                      "icon": <i
                        className="material-icons-outlined rotated"
                      >
                        grid_on
                      </i>,
                      "title": "Grid",
                    },
                    Object {
                      "icon": <i
                        className="material-icons-outlined rotated"
                      >
                        space_bar
                      </i>,
                      "title": "Spacing",
                    },
                  ],
                  "icon": <i
                    className="material-icons-outlined rotated"
                  >
                    wb_incandescent
                  </i>,
                  "title": "Get Started",
                },
                Object {
                  "children": Array [
                    Object {
                      "title": "Introduction",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "title": "Event 1",
                        },
                        Object {
                          "title": "Event 2",
                        },
                        Object {
                          "title": "Event 3",
                        },
                      ],
                      "lastParent": true,
                      "title": "Events",
                    },
                    Object {
                      "title": "Conclusion",
                    },
                  ],
                  "icon": <i
                    className="material-icons-outlined"
                  >
                    palette
                  </i>,
                  "lastParent": false,
                  "title": "Identity",
                },
              ]
            }
            sticky={true}
          />
        </div>
        <main
          className="col main-content border-left"
        >
          Main content things...
        </main>
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar TopbarWide renders 1`] = `
<Fragment>
  <h2
    id="topbar-wide"
  >
    Wide topbar
  </h2>
  <p>
    This is a wide topbar.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar-1"
      logout={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "name": "Menu",
          },
          "items": Array [
            Object {
              "icon": "home",
              "name": "Home",
            },
            Object {
              "icon": "shopping_cart",
              "name": "Purchase history",
            },
            Object {
              "icon": "settings",
              "name": "Settings",
            },
          ],
        }
      }
      wide="xl"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar TopbarWithSidebar renders 1`] = `
<Fragment>
  <h2
    id="topbar-with-sidebar"
  >
    Topbar with sidebar content
  </h2>
  <p>
    It is recommended to have the sidebar content in the topbar menu on smaller devices, such as phones and tablets (see more on sidebar 
    <Link
      to="/components/sidebar"
    >
      here
    </Link>
    ).
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar-sidebar"
      logout={true}
      sidebar={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "name": "Menu",
          },
          "items": Array [
            Object {
              "icon": "home",
              "name": "Home",
            },
            Object {
              "icon": "shopping_cart",
              "name": "Purchase history",
            },
            Object {
              "icon": "settings",
              "name": "Settings",
            },
          ],
        }
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    The topbar is used to give users an easily available navigational bar at the top of your web application.
  </p>
  <Overview />
  <TopbarWide />
  <TopbarSticky />
  <TopbarWithSidebar />
  <TopbarPNG />
  <JavascriptMethods />
</DocContainer>
`;
